File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    
    while progresses:
        
        for i, spd in enumerate(speeds):
            progresses[i] += spd
            
        cnt = 0
        
        while progresses and progresses[0] >= 100:
            del progresses[0], speeds[0]
            cnt += 1
        
        if cnt > 0:
            answer.append(cnt)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] res = new int[100];
        int day = 1;
        for (int i = 0; i < progresses.length; i++) {
            while (progresses[i] + speeds[i]*day < 100) {
                day++;
            }
            res[day]++;
        }
        return Arrays.stream(res).filter(i -> i != 0).toArray();
    }
}